server/src/controllers/surveyController.ts

```diff
@@ -12,17 +13,10 @@ export const storeSurveyResult = (surveyDBHandler: SurveyDBHandler) => async (
 ): Promise<any> => {
   try {
     const { body: requestData } = req;
-    const { callId, acsUserId, response } = requestData;
 
     // Validation.
-    // If any one of field does not exists in the payload the validation failed.
-    if (!callId || !acsUserId || response === undefined) {
-      const errors: string[] = [];
-
-      if (!callId) errors.push('callId is missing');
-      if (!acsUserId) errors.push('acsUserId is missing');
-      if (response === undefined) errors.push('response is missing');
-
+    const errors = surveyResultRequestValidator(requestData);
```

Comment:
should you also validate for the duplicate record, ignore if already saved?

Reply:
hmm in that case, how can we determine duplicate record? Should I check `callId` ? or should I check all of the field?

Reply:
Added the logic to check if the survey is already exists or not.